Windows salt-master failures due to passing 'log_queue'

skizunov opened the pull request:
On Windows, there are failures that appear like:
`TypeError: __init__() got an unexpected keyword argument 'log_queue'`

To fix this:

salt/utils/events.py:
- Objects inheriting from
  `salt.utils.process.SignalHandlingMultiprocessingProcess` must be
  capable of passing the kwarg 'log_queue' through to the parent class
  `MultiprocessingProcess` so that it will be used on Windows. See PR
  #26999 as to why this was needed.

salt/utils/reactor.py:
- Same as event.py
- Moved `__setstate__` and `__getstate__` from `salt.state.Compiler`
  to `salt.utils.reactor.Reactor`. The reason is that this is the most
  derived class, and we want to accomplish two things:
  1) Avoid pickling self.rend (inherited from salt.state.Compiler)
  which is unpicklable, hence the need for `__setstate__` and
  `__getstate__` in the first place.
  2) Initialize the class from the top-most point. The `__init__` function
  here will, in turn, initialize parent classes such as
  `salt.state.Compiler`, creating the renderer as needed.

salt/state.py:
- Removed `__setstate__` and `__getstate__` from `salt.state.Compiler`
  as per comment above.

Signed-off-by: Sergey Kizunov sergey.kizunov@ni.com

skizunov committed:
Windows salt-master failures due to passing 'log_queue'

On Windows, there are failures that appear like:
`TypeError: __init__() got an unexpected keyword argument 'log_queue'`

To fix this:

salt/utils/events.py:
- Objects inheriting from
`salt.utils.process.SignalHandlingMultiprocessingProcess` must be
capable of passing the kwarg 'log_queue' through to the parent class
`MultiprocessingProcess` so that it will be used on Windows. See PR
26999 as to why this was needed.

salt/utils/reactor.py:
- Same as event.py
- Moved `__setstate__` and `__getstate__` from `salt.state.Compiler`
to `salt.utils.reactor.Reactor`. The reason is that this is the most
derived class, and we want to accomplish two things:
1) Avoid pickling self.rend (inherited from salt.state.Compiler)
which is unpicklable, hence the need for `__setstate__` and
`__getstate__` in the first place.
2) Initialize the class from the top-most point. The `__init__` function
here will, in turn, initialize parent classes such as
`salt.state.Compiler`, creating the renderer as needed.

salt/state.py:
- Removed `__setstate__` and `__getstate__` from `salt.state.Compiler`
as per comment above.

Signed-off-by: Sergey Kizunov <sergey.kizunov@ni.com>

s0undt3ch commented:
Since the complain it about the `log_queue`, can we set that keyword argument instead using `**kwargs`? I think it's always wiser to be explicit....

skizunov committed:
Passing in `log_queue` explicitly instead of using `**kwargs`

salt/utils/process.py:
- Since `log_queue` is now passed in explicitly, it now may exist with a
value of `None`. `MultiprocessingProcess.__init__` has been modified to
handle this case.

Signed-off-by: Sergey Kizunov <sergey.kizunov@ni.com>

skizunov commented:
@s0undt3ch : I updated the code incorporating your feedback.

cachedout commented:
@skizunov Thanks. I will merge this after the tests run.

cachedout commented:
Merged. Thanks as always for the excellent work, @skizunov. It is very much appreciated.
